packages/amplify_datastore/ios/Classes/types/model/FlutterSerializedModel.swift

```diff
@@ -184,9 +208,51 @@ struct FlutterSerializedModel: Model, JSONValueHolder {
                     result[key] = [
                         "id": modelId,
                         "modelName": nextModelName,
-                        "serializedData": try generateSerializedJsonData(values: deserializedValue, flutterModelRegistration: flutterModelRegistration, modelName: nextModelName)
+                        "serializedData": try generateSerializedJsonData(
+                            values: deserializedValue,
+                            modelSchemaRegistry: modelSchemaRegistry,
+                            customTypeSchemaRegistry: customTypeSchemaRegistry,
+                            modelName: nextModelName
+                        )
+                    ]
+                }
+                // if embedded
+                else if case .embedded(_, .some(let customTypeSchema)) = field!.type {
+                    let customTypeName = customTypeSchema.name
+                    result[key] = [
+                        "customTypeName": customTypeName,
+                        "serializedData": try FlutterSerializedModel.generateSerializedJsonData(
+                            values: deserializedValue,
+                            modelSchemaRegistry: modelSchemaRegistry,
+                            customTypeSchemaRegistry: customTypeSchemaRegistry,
+                            modelName: customTypeName
+                        )
                     ]
                 }
+            } else if case .array(let jsonArray)  = value {
+                var modifiedArray:[Any?] = []
+                for item in jsonArray {
+                    // List of CustomType
+                    if case .object(let deserializedItem) = item,
+                       case .embeddedCollection(_, .some(let customTypeSchema)) = field?.type {
+                        let customTypeName = customTypeSchema.name
+                        let parsedItem = [
+                            "customTypeName": customTypeName,
+                            "serializedData": try FlutterSerializedModel.generateSerializedJsonData(
+                                values: deserializedItem,
+                                modelSchemaRegistry: modelSchemaRegistry,
+                                customTypeSchemaRegistry: customTypeSchemaRegistry,
+                                modelName: customTypeName
+                            )
+                        ] as [String : Any]
+                        modifiedArray.append(parsedItem)
+                    }
+                    // List of primitve types e.g. [String]
+                    else {
+                        modifiedArray.append(FlutterSerializedModel.deserializeValue(value: item, fieldType: nil))
+                    }
+                }
```

Comment:
This part of code looks duplicate as the part in `generateSerializedData`. This part will be removed after merging https://github.com/aws-amplify/amplify-flutter/pull/806